### PR TITLE
fix: use new contract interface

### PR DESCRIPTION
**Pull request by janek26**
merge after https://github.com/0xs34n/starknet.js/pull/138 gets merged, released and the dependency of argent updated

### CODE REVIEW

**Review by delaaxe — APPROVED** [review]

**Comment by delaaxe** [comment]
Should the dapp be updated too?

**Comment by janek26** [comment]
pushed a very tiny update.
the rest seems to be fine as we don't use the Contract interface

**Comment by delaaxe** [comment]
Can't we use this interface for mint and transfer?

**Comment by janek26** [comment]
yes, and thanks for suggesting
catchen some ts type bugs in the process: https://github.com/0xs34n/starknet.js/pull/144

https://github.com/0xs34n/starknet.js/pull/144 needs to be merged, released and the dependency of argent updated before merge

File: packages/dapp/src/services/token.service.ts [hunk 1]
```diff
@@ -32,16 +34,17 @@ export const mintToken = async (
     throw Error("starknet wallet not connected")
   }
 
-  const receiver = number.toBN(activeAccount).toString()
-  const amount = getUint256CalldataFromBN(
-    utils.parseUnits(mintAmount, 18).toString(),
+  console.log(starknet.account)
```
Comment (delaaxe) [inline packages/dapp/src/services/token.service.ts:37]: console.log

File: packages/dapp/src/services/token.service.ts [hunk 2]
```diff
@@ -32,16 +34,17 @@ export const mintToken = async (
...
+  const erc20Contract = new Contract(
+    Erc20Abi as any,
```
Comment (delaaxe) [inline packages/dapp/src/services/token.service.ts:39]: Don't use any in a demo code that people copy and paste into their own products

File: packages/dapp/src/services/token.service.ts [hunk 3]
```diff
@@ -32,16 +34,17 @@ export const mintToken = async (
...
   )
 
-  return await starknet.account.execute({
-    contractAddress: erc20TokenAddressByNetwork[network],
-    entrypoint: "mint",
-    calldata: stark.compileCalldata({ receiver, amount }),
-  })
+  return erc20Contract.mint(
+    activeAccount,
+    getUint256CalldataFromBN(utils.parseUnits(mintAmount, 18).toString()),
```
Comment (delaaxe) [inline packages/dapp/src/services/token.service.ts:46]: This whole rigmarole could be abstracted away

File: packages/extension/src/inpage/index.ts [hunk 4]
```diff
@@ -228,6 +228,8 @@ export class ArgentXAccount extends Account {
         }),
     ])
 
+    console.log(result)
```
Comment (delaaxe) [inline packages/extension/src/inpage/index.ts:231]: console.log

**Review by delaaxe — CHANGES_REQUESTED** [review]

**Review by delaaxe — APPROVED** [review]

**Pull request merged**